### ISSUE TITLE
Hide spoilery tags

### ISSUE THREAD

**Issue by alice-blue** [iftechfoundation/ifdb-suggestion-tracker#82]
Some tags may give away surprises in a game. There are different ways these could be dealt with---here are a few:

* Add the ability to mark certain tags as spoilers for a given game. When viewing a game page, the list of spoiler-marked tags would be hidden by default, so you would have to click to see them.

* Hide _all_ tags by default, until you click to see them.

* Hide certain categories of tags by default. So, for example, "CYOA" doesn't get hidden, because that's about the format of the game, but anything related to the content of the game (for example, "dream" or "drowning" or "alien protagonist") gets hidden until you click.

**Comment by qdacsvx** [comment]
-1

1. I like to add tags. It will be inconvenient if any tags are hidden by default.
2. Users should flag offensive tags. It will be inconvenient if any tags are hidden by default.

I can find games and play games without seeing any tags. Why do you need to hide tags?

>Add the ability to mark certain tags as spoilers for a given game.

I think usually a spoiler tag will be a spoiler for every game. So you only need a list of the spoiler tags - you don't need one per game.

I think of the alternatives of a personal list of spoiler tags or a moderater-controlled public list (moderators dictate public tag visibility) I prefer the personal list of spoiler tags.

The tag list edit page would be a page of tags with checkboxes to include them. There could be a menu to select the list to edit. There could be multiple pages if needed. There should be a compact representation which is just the tags in a comma list. This could be shown at the top of the tag list edit page so you can quickly track the list. Tags should be clickable links (tag search) in the compact list.

Users could optionally share their personal spoiler tag lists so it's easy to get started by copying others. A privileged user could pick a shared user list (users could vote for shared lists like user CSS) and select it to fill in the default list so every new user gets a list by default. The picked list should display a badge for the distinction. It should be easy for users to opt out of spoiler blocking. Users might like to use tag lists for alternative purposes like hiding annoying tags.

>Hide certain categories of tags

Someone will need to add/remove tags from these categories.
These categories could also be personal tag lists that the user can add/remove tags.
There could be an automatic list of new tags since the user last logged in. So you can update your spoiler tags / annoying tags. There could be a public page showing new tags "Hot tags" #86 .

Users might like to create and name (and optionally share) several personal tag lists. Or there could be preset named lists that users ask for like "Annoying block" "Spoiler block" "Spoiley ok" "ok"...
Some users might care about certain types of spoilers.

There should be a way to show/hide these tag list/categories (button could open a menu) when you are on a game page viewing the tags.

**Comment by alice-blue** [comment]
There are different groups of people who might like the option of having spoilery tags hidden:

- People who aren't trying to look at tags, but who might see them by mistake before playing a game (for example, when scrolling past the tags on the way to look at reviews)

- People who look at tags before playing a game to help them decide if they want to play it

- People adding tags who aren't sure whether to add a certain tag since it may spoil the game for someone else

If users had the ability to mark tags as spoilers for a game, there could be a setting in the user profile for hiding or showing all spoilery tags. Or people could choose which spoilery tags to hide (because there may be some tags a player always wants to know about, even if they are spoilers in a given game).

One potential challenge of this is major vs. minor spoilers.  I imagine many people would want major spoilers hidden (for example, you find out in the last scene that the player character is actually a villain). Not sure about minor ones (for example, you find out 1/4 of the way into the game that player character is a villain). I'm not sure if it would be too much granularity to be able to mark tags as "major spoiler" or "minor spoiler."

**Comment by qdacsvx** [comment]
My disagreement is mainly that in the options to deal with spoilers, some options imply that some tags would be hidden for everyone even if they wanted to see all tags.

>If users had the ability to mark tags as spoilers for a game, there could be a setting in the user profile for hiding or showing all spoilery tags.

This option (showing all spoilery tags) seems like it would be satisfactory to me if it would show all tags. @alice-blue can you add such an option (to show all tags) to the first post? Then I won't need to disagree.

Would users share these categories of tags so you don't have to mark every spoilery tag yourself, as long as someone marked it, it will be hidden for you (and for the public, presumably)?

What if someone marks a non-spoiler tag as a spoiler by mistake?

>I'm not sure if it would be too much granularity to be able to mark tags as "major spoiler" or "minor spoiler."

I think major spoilers in tags should be removed from IFDB. I think there are only one or two cases right now.

It's more flexible to write spoilers in a review where you can warn people what is in the spoiler and there is space for longer spoilers.

Spoiler tags that are commonly hidden won't be reviewed for flagging offensive content often. Perhaps there will be a culture to view spoilers after you finish playing.

**Comment by alice-blue** [comment]
> can you add such an option (to show all tags) to the first post?

Done.

> Would users share these categories of tags so you don't have to mark every spoilery tag yourself, as long as someone marked it, it will be hidden for you (and for the public, presumably)?

If you're talking about marking a tag spoilery for a given game, yes. The people who have played the game (or authored it) are the ones who will know what the spoilers are, and will be able to mark the tags as spoilers. Ideally this would be done when the tag is first added to the game. The people who haven't played the game yet are the ones who would want to avoid seeing the spoilers, but won't know what they are.

> What if someone marks a non-spoiler tag as a spoiler by mistake?

Depending on how much control is ultimately given to users vs. mods for tagging, it could be that anyone could fix it, or the tagger could fix it, or someone could flag it for a mod to fix.

**Comment by curiousdannii** [comment]
MyDramaList is a site that lets you add tags marked as spoilers. Seems to work well IMO.

**Comment by qdacsvx** [comment]
@curiousdannii does the author of a work often add it to [MyDramaList](https://mydramalist.com/)? For IFDB many works are added by new users who aren't familiar with the tagging system. Some added works get no tags, others get novel tags that don't match the common tags.

@alice-blue Consider the demand for users to hide annoying tags #6 . If it takes a week for a moderator to remove a tag #71 , but I can hide it by marking it as a spoiler right now, there is a perverse incentive to falsely mark nonspoilers as spoilers. I fear that some tags will be hidden most of the time.

**Comment by curiousdannii** [comment]
No, it would all be fans tagging things at MyDramaList. It has another feature which helps with tag disagreements: you can vote on the relevance of a tag, rather than it simply being a binary thing. That would be a huge change to make to IFDB however.

**Comment by qdacsvx** [comment]
[off topic]
I think bipolar voting would favour proposers for visible tags but conservatives in general. This is because proposers can organise to outnumber conservatives for any particular tag, (conservatives are spread thinly across all tags) but proposers mainly care about visible tags for works on the front page or The Top 100. So conservatives quietly voting everywhere will control most tags on IFDB.

[tag spoilers]
Another option to avoid tags spoiling people is to remove spoiling tags from IFDB #96 .

I notice that if it is not allowed to submit spoilers as tags, although there are alternative ways to submit spoilers e.g. #56, there would be no way to submit a spoiler to IFDB anonymously. Any suggestions?

**Comment by qdacsvx** [comment]
I will have a stab at it. @alice-blue does this sound good to you? Feel free to add your ideas.

There is nowhere to put content spoilers. Editing the game description is slow. Making tags is fast. So content spoilers get dumped in tags. But tags are public so anybody can be spoiled.

There should be a special place to write content spoilers. They should be hidden by default. To avoid misuse, spoilers should be 100% independent of tags.

tags x y z
My tags: twine

spoilers (click to show)
My spoilers: multiple endings

Spoilers should be as prominent as tags because people sometimes want to read them.
Spoilers could be rated for how severe on a 1-5 scale, or a scale in words like minor,major...
Revealing would only show 1 level; you would click again to reveal the next level. There could be tabs/links for different spoiler topics. Each topic would have it's own levels of spoilers.
The topics could be preset or freely named by users for specific purposes "endings" "plot" "protagonist"...

There could be a public "common spoilers" page for people who want to see all the spoilers in use.

**Comment by alice-blue** [comment]
Tags are easily searchable.

If people are writing games, or doing research, and they want to see how a certain topic or puzzle type or character type has been handled in the past, they can click the tag to see a list of all the relevant games. They may even want to know about the ones where that topic/puzzle type/character type is a spoiler. (And it would be nice if the search results let you choose whether to see the spoilery results or not.)

There has also been discussion about the possibility of people using tags to filter what shows up in their searches. So "If a game contains XYZ (regardless of whether XYZ is a spoiler in that game), I don't want it to show up in my search." Or a profile setting that says "I don't want a game that contains XYZ to show up in any of my searches, ever."

If IFDB users as a whole just don't want spoilers in the tags at all, and there are other ways to address these things, that's fine too.

It sounds like your suggestion is aimed more at people who just like to keep track of spoilers.
